### PR TITLE
Openaccess unpaywall link

### PR DESCRIPTION
**Pull request by bserna-usgs**

### CODE REVIEW

File: server/pubs_ui/pubswh/views.py [hunk 1]
```diff
@@ -272,6 +272,7 @@ def publication(index_id):
     altmetric_links = {'image': small_badge, 'details': pub_altmetric_details}
     pubdata['altmetric'] = altmetric_links
     crossref_data = get_crossref_data(pubdata.get('doi'))
+    unpaywall_data = get_unpaywall_data(pubdata.get('doi'))
```
Comment (mbucknell) [inline server/pubs_ui/pubswh/views.py]: This should be removed. The call to munge_data_for_display on line 269 fills in this information

File: server/pubs_ui/pubswh/tests/test_data.py [hunk 2]
```diff
@@ -1590,3 +1590,73 @@
     'message-version': '1.0.0',
     'status': 'ok'
 }
+
+unpaywall_200_ok = {'best_oa_location': {'evidence': 'open (via page says license)',
```
Comment (mbucknell) [inline server/pubs_ui/pubswh/tests/test_data.py]: Codacy (our code quality application) is complaining about the formatting here. You can see the complete report by clicking the Codacy link in the PR.

File: server/pubs_ui/pubswh/utils.py [hunk 3]
```diff
@@ -1051,6 +1072,26 @@ def get_crossref_data(doi, endpoint=CROSSREF_ENDPOINT, verify=VERIFY_CERT):
     return crossref_data
 
 
+@cache.memoize(timeout=2592000)  # Cache data for a month so the nice people at unpaywall yell at us
+def get_unpaywall_data(doi, endpoint=UNPAYWALL_ENDPOINT, verify=VERIFY_CERT):
```
Comment (mbucknell) [inline server/pubs_ui/pubswh/utils.py]: You can remove the verify parameter as it is unused.

File: server/pubs_ui/pubswh/views.py (hunk 1 above)
Comment (bserna-usgs) [reply server/pubs_ui/pubswh/views.py]: Thanks Mary, I just tested this and it works. I could have sworn I was having issues without this earlier

File: server/pubs_ui/pubswh/templates/pubswh/publication.html [hunk 4]
```diff
@@ -334,6 +335,15 @@ <h4>Links</h4>
                                                                href="{{ pubdata['offers']['schema:offers']['schema:url'] }}"><strong>USGS
                                 Store</strong></a></li>
                     {% endif %}
+                    {% if pubdata['isOA'] == True %}
```
[the remark this answers is not in the document]
Comment (bserna-usgs) [reply server/pubs_ui/pubswh/templates/pubswh/publication.html]: Great suggestion!

File: server/pubs_ui/pubswh/utils.py [hunk 5]
```diff
@@ -687,6 +689,25 @@ def has_excel(pubdata):
     return pubdata
 
 
+def has_oa_link(pubdata):
```
[the remark this answers is not in the document]
Comment (bserna-usgs) [reply server/pubs_ui/pubswh/utils.py]: 👍

File: server/pubs_ui/pubswh/utils.py [hunk 6]
```diff
@@ -687,6 +689,25 @@ def has_excel(pubdata):
...
+    """
+    Function to identify if a pub contains an open access link from unpaywall.
+
+    If there is an open access link, the link and the host type of the link is appended to the pubdata.
+    :param pubdata:
+    :return: pubdata
+    """
+    pubdata['isOA'] = False
```
[the remark this answers is not in the document]
Comment (bserna-usgs) [reply server/pubs_ui/pubswh/utils.py]: That sounds better, can you make sure my next commit looks like what you had in mind? Thanks!

**Comment by bserna-usgs** [comment]
@jkreft-usgs can you take a look at this

**Comment by jkreft-usgs** [comment]
it looks like the Codacy things are basically red herrings (test data with a long line, and a snake case variable that it wants to be uppercase for some reason) that we can ignore

**Pull request merged**